Dotty fails to to infer type params for Java Stream API

``` scala

object T{
  import java.util.ArrayList, java.util.stream.{Stream => JStream}
  new java.util.ArrayList[String]().stream.map(_.toInt).map(_.toString): JStream[String]
}
```

```
Streams.scala:3: error: toInt is not a member of <unknown instance of type java.util.function.Function[_ >: String, _ = Nothing]>
  .
java$util$function$Function$$T(_$1)
  new java.util.ArrayList[String]().stream.map(_.toInt).map(_.toString): JStream[String]
                                                 ^
Streams.scala:3: error: type mismatch:
 found   : <unknown instance of type java.util.function.Function[_ >: String, _ = Nothing]>
  .
java$util$function$Function$$T =>
  <unknown instance of type
    java.util.function.Function[_ >: String, _ = Nothing]
  >.java$util$function$Function$$R
 required: java.util.function.Function[_ >: String, _ <: Nothing']
  new java.util.ArrayList[String]().stream.map(_.toInt).map(_.toString): JStream[String]
                                               ^
Streams.scala:3: error: type mismatch:
 found   : String
 required: <unknown instance of type java.util.function.Function[_, _ <: String]>.
  java$util$function$Function$$R
  new java.util.ArrayList[String]().stream.map(_.toInt).map(_.toString): JStream[String]
                                                            ^
three errors found
```

see also https://groups.google.com/forum/#!msg/scala-language/Xy29pcRE04c/rwCvztHjsP8J for same issue for scalac

--- a/tests/pos/i801.scala
+++ b/tests/pos/i801.scala
@@ -1,0 +1,9 @@
+object T1 {
+  import java.util.ArrayList, java.util.stream.{Stream => JStream}
+  new java.util.ArrayList[String]().stream.map(_.toInt).map(_.toString): JStream[String]
+}
+
+object T2 {
+  import java.util._, java.util.stream.{Stream => JStream}
+  def f: JStream[String] = new java.util.ArrayList[String](Arrays.asList("1", "2")).stream.map(_.toInt).map(_.toString)
+}

